src/Components/News/Detail/NewsDetailLayout.js

```diff
@@ -1,35 +1,44 @@
-import React, { useEffect,useState } from 'react';
-import { useParams } from 'react-router';
-import NewsTitle from './NewsTittle';
-import { Box } from '@mui/material';
-import { getNewById } from '../../../Services/NewServices/getNewById';
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router";
+import NewsTitle from "./NewsTittle";
+import { Box } from "@mui/material";
+import { getNewById } from "../../../Services/NewServices/getNewById";
+import LoadingSpinner from "../../../Utils/loadingSpinner";
+import "../../../Styles/CardStyle.css";
 
-const NewsDetailLayout = ({prop}) => {
+const NewsDetailLayout = ({ prop }) => {
+  const [newData, setNewData] = useState({});
+  const [loading, setIsLoading] = useState(true);
 
-    const [newData,setNewData] = useState({})
+  const { id } = useParams();
 
-    const {id} = useParams()
+  const loadNewData = async (id) => {
+    const { data } = await getNewById(id);
+    data && setNewData(data);
+  };
 
-    const loadNewData = async () => {
-        const {data} = await getNewById()
-        setNewData(data)
-    }
+  useEffect(() => {
+    newData.image && setIsLoading(false);
```

Comment:
and what happens if data on loadNewData do not have data, for example, the id do not exists. It will never stop loading and that will be a problem. you should setIsLoading as false on loadNewData call.

Reply:
Done! I passed it to the useEffect that executes loadNewData which depends on the existence of an ID to execute but I consider this to be fine because if there was no id, the getNewById of the service would already be firing an alert by mistake.